### ISSUE TITLE
[Bug][a11y]: SideNav should express hierarchy using list and listitem

### ISSUE THREAD

**Issue by majornista** [adobe/spectrum-web-components#3348]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

SideNav, SideNavHeading, SideNavItem

### Expected behavior

1. SideNav should support a `label` and `labelledby` so that the `<nav>` element in the shadowDom can be distinguished from any other `nav` elements in a web page.
2. SideNav should wrap its `slot[name="descendant"]` element in a `div[role="list"]`, so that SideNav items will be contained within a hierarchical list.
3. SideNavHeading should have a default `role` property of `listitem` :
```ts
    @property({ reflect: true })
    public role = 'listitem';
```
4. SideNavHeading shadowDom element with `id="list"` should also have `role="list"`, so that it the list hierarchy will be appropriately communicated to assistive technology.
```ts
    protected override render(): TemplateResult {
        return html`
            <h2 id="heading">${this.label}</h2>
            <div id="list" aria-labelledby="heading" role="list">
                <slot name="descendant"></slot>
            </div>
        `;
    }
```
5. SideNavItem should have a default `role` property of `listitem` :
```ts
    @property({ reflect: true })
    public role = 'listitem';
```
6. SideNavItem should wrap its `slot[name="descendant"]` element in a `div[role="list"]`, so that descendant SideNavItems will be contained within a hierarchical list, and the SideNavItem link in shadowDom should include `aria-expanded` and `aria-controls` when `this.hasChildren === true` to express the current state of the expandable node:
```ts
    protected override render(): TemplateResult {
        return html`
            <a
                href=${this.href || '#'}
                target=${ifDefined(this.target)}
                download=${ifDefined(this.download)}
                rel=${ifDefined(this.rel)}
                data-level="${this.depth}"
                @click="${this.handleClick}"
                id="item-link"
                aria-current=${ifDefined(
                    this.selected && this.href ? 'page' : undefined
                )}
                aria-expanded=${ifDefined(
                    this.hasChildren ? this.expanded : undefined
                )}
                aria-controls=${ifDefined(
                    this.hasChildren && this.expanded ? 'list' : undefined
                )}
            >
                <slot name="icon"></slot>
                ${this.label}
                <slot></slot>
            </a>
            ${this.expanded
                ? html`
                    <div id="list" aria-labelledby="item-link" role="list">
                        <slot name="descendant"></slot>
                    </div>
                  `
                : html``}
        `;
    }
```
 7. The SideNavItem link should remain tabbable/focusable when it is collapsed, so that a user can toggle the expanded/collapsed state of descendant list.
 ```ts
     protected override updated(changes: PropertyValues): void {
        if (this.hasChildren && this.expanded && !this.selected) {
            this.focusElement.tabIndex = 0;
        } else {
            this.focusElement.removeAttribute('tabindex');
        }
        super.updated(changes);
    }
 ```

### Actual behavior

SideNav does not communicate hierarchy that is communicated visually, to users of assistive technology.

### How can we reproduce this issue?

3. Scroll to '....'
4. Check console
5. See error
